kinto/core/storage/memory.py

```diff
@@ -332,26 +341,54 @@ def apply_filters(records, filters):
                 if isinstance(right, int):
                     right = str(right)
 
-            left = find_nested_value(record, f.field)
+            left = find_nested_value(record, f.field, MISSING)
 
             if f.operator in (COMPARISON.IN, COMPARISON.EXCLUDE):
                 right, left = left, right
+            elif f.operator not in (COMPARISON.LIKE, COMPARISON.HAS):
+                left = schwartz_transform(left)
+                right = schwartz_transform(right)
+
+            if f.operator == COMPARISON.HAS:
+                matches = left != MISSING if f.value else left == MISSING
             else:
-                if left is None:
-                    right_is_number = (
-                        isinstance(right, (int, float)) and
-                        right not in (True, False))
-                    if right_is_number:
-                        # Python3 cannot compare None to a number.
-                        matches = False
-                        continue
-                    else:
-                        left = ''  # To mimic what we do for postgresql.
-            matches = matches and operators[f.operator](left, right)
+                matches = matches and operators[f.operator](left, right)
         if matches:
             yield record
 
 
+def schwartz_transform(value):
```

Comment:
cool, did not know about this name :) 

btw, on wikipedia it's called schwartzian transform